src/Elems/itemList.js

```diff
@@ -0,0 +1,210 @@
+        try {
+            const res = await axios.delete('http://localhost:1234/products/'+_id+'/delete');
+            if (res.status == 200){
+                this.props.data.splice(itemIndex, 1); 
+                toast.error('You delete todo');               
+            }; 
+        }
+        catch (err) {
+        };
+        this.setState({data:this.props.data});
+    };
+    
+    onUndoneListBtnClick = () => {
+        this.setState({filterStatus:'undone'})
+    };
+
+    onDoneListBtnClick = () => {
+        this.setState({filterStatus:'done'})
+    };
+    
+    onAllListBtnClick = () => {
+        this.setState({filterStatus:'all'})
+    };
+
+    onMarkAllBtnClick = () => {
+        if (this.state.markAll == true){
+           this.onMarkAllDoneBtnClick()
+        }
+        else this.onMarkAllUndoneBtnClick();
+    };
+
+    onMarkAllDoneBtnClick = async () => {
+        try {
+            const res = await axios.put('http://localhost:1234/products/change',{
+                status:true
+            });
+                if (res.status == 200){
+                    return this.props.data.map((item) => {
+                    return item.status = true;
+                    }),
+                    this.setState({data:this.props.data, markAll:false}),
+                    toast.info('You marcked all as "done"');
+                };
+            }
+            catch (err) {
+            };
+    };
+
+    onMarkAllUndoneBtnClick = async () => {
+        try {
+            const res = await axios.put('http://localhost:1234/products/change',{
+                status:false
+            })
+                if (res.status == 200){
+                    return  this.props.data.map((item) => {
+                                return item.status = false
+                            }),
+                    this.setState({data:this.props.data, markAll:true}),
+                    toast.info('You marcked all as "undone"');
+                };
+        }
+        catch (err) {
+        };
+    }
+    
+    onDeleteAllDoneBtnClick = () => {
+        this.props.deleteAllDone();
+        toast.warning('You deleted all "done" todos');    
+    };
+    
+    status_All_buttons = () => {
+        if (this.state.filterStatus == 'all') {
+            return ( 
+                this.state.allChkd = 'this',
+                this.state.doneChkd = '',
+                this.state.undoneChkd =''
+                )
+        }
+        else if (this.state.filterStatus == 'done') {
+            return (
+                this.state.allChkd = '',
+                this.state.doneChkd = 'this',
+                this.state.undoneChkd =''
+                )
+        }
+        else if (this.state.filterStatus == 'undone') {
+            return (
+                this.state.allChkd = '',
+                this.state.doneChkd = '',
+                this.state.undoneChkd ='this'
+            )
+        }
+    }
```

Comment:
Need to find another way to filter data + need to define states inside this function to the state of the component